development/zh-cn/e2e-test.md

```diff
@@ -0,0 +1,191 @@
+            return tab.cast(new WorkerGroupPage(driver));
+        }
+        if (tab == QueuePage.class) {
+            menuQueueManage().click();
+            return tab.cast(new QueuePage(driver));
+        }
+        throw new UnsupportedOperationException("Unknown tab: " + tab.getName());
+    }
+```
+
+![SecurityPage](../../img/e2e-test/SecurityPage.png)
+
+对于导航栏选项的跳转，在`org/apache/dolphinscheduler/e2e/pages/common/NavBarPage.java` 中提供了 goToNav 的方法。当前支持的页面为：项目管理（ProjectPage）、安全中心（SecurityPage）和资源中心（ResourcePage）。
+
+```java
+    public <T extends NavBarItem> T goToNav(Class<T> nav) {
+        if (nav == ProjectPage.class) {
+            WebElement projectTabElement = new WebDriverWait(driver, 60)
+                .until(ExpectedConditions.elementToBeClickable(projectTab));
+            ((JavascriptExecutor)driver).executeScript("arguments[0].click();", projectTabElement);
+            return nav.cast(new ProjectPage(driver));
+        }
+
+        if (nav == SecurityPage.class) {
+            WebElement securityTabElement = new WebDriverWait(driver, 60)
+                .until(ExpectedConditions.elementToBeClickable(securityTab));
+            ((JavascriptExecutor)driver).executeScript("arguments[0].click();", securityTabElement);
+            return nav.cast(new SecurityPage(driver));
+        }
+
+        if (nav == ResourcePage.class) {
+            WebElement resourceTabElement = new WebDriverWait(driver, 60)
+                .until(ExpectedConditions.elementToBeClickable(resourceTab));
+            ((JavascriptExecutor)driver).executeScript("arguments[0].click();", resourceTabElement);
+            return nav.cast(new ResourcePage(driver));
+        }
+
+        throw new UnsupportedOperationException("Unknown nav bar");
+    }
+```
+
+### 2、E2E-Cases
+
+当前所支持的 E2E 测试案例，主要包括：文件管理、项目管理、队列管理、租户管理、用户管理、Worker 分组管理和工作流测试。
+
+![E2E_Cases](../../img/e2e-test/E2E_Cases.png)
+
+下面以租户管理测试为例，前文以及说明，我们使用 docker-compose 进行部署，所以每个测试案例，都需要以注解的形式引入对应的文件。
+
+使用 Selenium 所提供的 RemoteWebDriver 来加载浏览器。在每个测试案例开始之前都需要进行一些准备工作。比如：登录用户、跳转到对应的页面（根据具体的测试案例而定）。
+
+```java
+    @BeforeAll
+    public static void setup() {
+        new LoginPage(browser)
+                .login("admin", "dolphinscheduler123") // 登录进入租户界面
+                .goToNav(SecurityPage.class) // 安全中心
+                .goToTab(TenantPage.class)
+        ;
+    }
+```
+
+在完成准备工作之后，就是正式的测试案例编写。我们使用 @Order() 注解的形式，用于模块化，确认测试顺序。在进行测试之后，使用断言来判断测试是否成功，如果断言返回 true，则表示创建租户成功。可参考创建租户的测试代码：
+
+```java
+    @Test
+    @Order(10)
+    void testCreateTenant() {
+        final TenantPage page = new TenantPage(browser);
+        page.create(tenant);
+
+        await().untilAsserted(() -> assertThat(page.tenantList())
+                .as("Tenant list should contain newly-created tenant")
+                .extracting(WebElement::getText)
+                .anyMatch(it -> it.contains(tenant)));
+    }
+```
+
+其余的都是类似的情况，可参考具体的源码来理解。
```

Comment:
Should we copy-paste the link of E2E test case https://github.com/apache/dolphinscheduler/tree/dev/dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases ?

Reply:
Good idea!